packages/astro/src/content/config.ts

```diff
@@ -80,22 +103,10 @@ export type LiveCollectionConfig<
 	loader: L;
 };
 
-type LoaderConstraint<TData extends { id: string }> =
```

Comment:
The loader constraint is important, this is what allowed the loader API breaking change